### DISCUSSION TITLE
Eratic Y axis behavior

### DISCUSSION THREAD

**Discussion by odataway** [grblHAL/core#306] [Q&A]
I'm using a teensy 4.1 (MIMXRT1062), grblhal, the T41U5XBB breakout board, DM542T drivers, NEMA 23 motors on a three axis machine. Board is powered via USB (laptop). Used the WebBuilder to create the image for the controller, basic three axis machine. 

When I fire up the machine I have normal operation and homing, shortly thereafter I'll get erratic behavior on the Y axis (turning in the same direction at all times). Drivers are known to be good, on the BoB y axis direction pin I get unstable voltages. On the X and Z axis I get 0 or 4.75 volts depending on jog direction which appears normal. On the Y axis direction pin I'll get quick bursts of a few volts, and then back to zero when moving in one direction, zero in the other direction. In other words I'm not getting a consistent +voltage at the Y axis direction pin. 

For some history ... I had to replace the Y axis driver due to damage from me turning the motor by hand while it was plugged in and powered up ( I have since learned not to do that). Is it possible that I have damaged the BoB Y axis? I have swapped drivers etc and the error follows the BoB Y axis. 

The fact that it works normally for the first few minute leads me to believe it's a heat related issue on a component of the BoB. Possibly damaged by my manually moving the stepper motors.

I have not removed the BoB and checked traces or anything yet, I do get the same voltage readings at either the terminal pin, or the through hole on board so I don't believe it's a soldering issue at the terminal block. grbl setup is identical for all axis.

Thanks for any advice.

**Comment by terjeio** [comment]
Try with firmware compiled for ganged Y-axis and connect the Y driver to the second motor output?

**Reply by odataway** [reply]
Update: Terjeio ... appreciate your quick response. Checked the voltage at the Y axis direction pin with the driver disconnected and it shows proper voltage changes ... so, despite my previous statement that the driver was "known good" that doesn't seem to be the case. It appears the BoB is seeing something on the Y axis direction pin it doesn't like when it's connected to the driver and is defaulting to zero volts?

I'm going to obtain a new driver and test again.

**Reply by odataway** [reply]
Just wanted to update. Replaced the Y axis driver and the machine appears to be responding normally. So between my ham handed treatment of the drivers/motors and the occasional use of UGS instead of IOSender I think I have moved on to the next step of testing.
Thank you.
